### ISSUE TITLE
Response headers on error callback

### ISSUE THREAD

**Issue by JZubero** [thelinmichael/spotify-web-api-node#63]
At the moment there is no possibility to fetch the `Retry-After` response header if there was an error 429 (rate limiting). This information would be needed to develop a resume/retry logic.

**Comment by JMPerez** [comment]
@JZubero You are right, the headers are only exposed when the request succeeds, but in this case it would make sense to also expose them on error.

**Comment by JZubero** [comment]
Indeed, that would be **extremely** helpful. Looking forward to this change! :+1:

**Comment by dandv** [comment]
:+1: In twitter-lite, we also [expose the response headers](https://github.com/Preposterous/twitter-lite/blob/1c9154f79f638af2303ff4f45165e83adbbbaef6/twitter.js#L58-L70).

**Comment by patrick-motard** [comment]
Most of the issues raised in this repo could probably be resolved by this. It seems in most cases users get a generic error back and have nothing to indicate to them how to fix the issue.
